src/Connection/PhpiredisSocketConnection.php

```diff
@@ -308,7 +310,10 @@ public function connect()
     {
         if (parent::connect() && $this->initCommands) {
             foreach ($this->initCommands as $command) {
-                $this->executeCommand($command);
+                $response = $this->executeCommand($command);
+                if ($response instanceof ResponseInterface && $response instanceof ErrorResponseInterface) {
+                    $this->onConnectionError($command->parseResponse($response), 0);
```

Comment:
How should I handle this then? Any chance you take over the PR if it's just a couple changes? I am not so familiar with the predis internals.